src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsSearch.js

```diff
@@ -24,39 +24,56 @@ export default class ItemsSearch extends Component {
     clearResults();
   };
 
-  executeSearch = (queryString) => {
+  hitAvailableForCheckout = (hits) =>
+    !_isEmpty(hits) &&
+    hits.length === 1 &&
+    hits[0].metadata.status === 'CAN_CIRCULATE';
+
+  checkoutItem = async (hit) => {
+    const {
+      checkoutItem,
+      patronDetails: { user_pid },
+    } = this.props;
+
+    const documentPid = hit.metadata.document.pid;
+    const itemPid = {
+      type: recordToPidType(hit),
+      value: hit.metadata.pid,
+    };
+    await checkoutItem(documentPid, itemPid, user_pid, true);
+  };
+
+  executeSearch = async (queryString) => {
     const { queryString: propsQueryString, fetchItems } = this.props;
     queryString = queryString || propsQueryString;
     // eslint-disable-next-line react/no-unused-state
-    this.setState({ prevSearchQuery: queryString, executedSearch: true });
-    fetchItems(queryString);
+    this.setState({ prevSearchQuery: queryString, executedSearch: true }, () =>
+      fetchItems(queryString)
+    );
   };
 
   onPasteHandler = async (event) => {
-    const { checkoutItem, patronDetails } = this.props;
     const { prevSearchQuery } = this.state;
+
     let queryString = event.clipboardData.getData('Text');
+
     const sameQueryString = prevSearchQuery === queryString;
 
     if (queryString && !sameQueryString) {
       await this.executeSearch(queryString);
-      const {
-        items: { hits },
-      } = this.props;
-
-      const hasOneHit =
-        !_isEmpty(hits) &&
-        hits.length === 1 &&
-        hits[0].metadata.status === 'CAN_CIRCULATE';
-
-      if (hasOneHit) {
-        const documentPid = hits[0].metadata.document.pid;
-        const itemPid = {
-          type: recordToPidType(hits[0]),
-          value: hits[0].metadata.pid,
-        };
-        await checkoutItem(documentPid, itemPid, patronDetails.user_pid, true);
-      }
+
+      setTimeout(async () => {
+        const {
+          items: { hits },
+        } = this.props;
+
+        const hasOneHit = this.hitAvailableForCheckout(hits);
+
+        if (hasOneHit) {
+          await this.checkoutItem(hits[0]);
+        }
+      }, 600);
```

Comment:
We cannot put a timeout because, in this case, it highly depends on the time that the HTTP request will take to fetch the data. We cannot control it.
In general, a timeout is, in my opinion, a code smell.

After having checked the implementation of the original React component, it is clear to me that it has not been implemented in the correct way.
What you have done in your previous implementation was more correct: my comment was referring to the fact that it is probably not ideal to add a `checkout` instead of a `fetchItems` method.

Suggestion to fix it. In Redux action, check if [fetchItems](https://github.com/inveniosoftware/react-invenio-app-ils/blob/master/src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/state/actions.js#L10) is called/reused somewhere else.

## if already called

- create a new action e.g. `fetchAndCheckoutIfOne` (please choose a better name :)) that will call the other `fetchItems` and then perform the checkout (move the checkout code to the action)

## if not already called

- as your previous implementation, move the checkout inside and then rename the method, so that you are sure that on other component in the future will use that function to just fetch items (and will have as side effect to checkout).